File: selection_sort.py
def find_smallest_number (numbers):
    smallest_number = numbers[0]
    index_smallest_number = 0
    for i in range (1, len(numbers)):
        if (numbers[i] < smallest_number):
            smallest_number = numbers[i]
            index_smallest_number = i
    return index_smallest_number

def selection_sort(numbers):
    sorted_numbers = []
    for i in range(len(numbers)):
        index_smallest_number = find_smallest_number(numbers)
        sorted_numbers.append(numbers.pop(index_smallest_number))
    return sorted_numbers

numbers = [2, -2, 10, 0]
print(selection_sort(numbers))

File: selection_sort.cpp
#include <vector>
#include <iostream>
#include <algorithm>

void selection_sort(std::vector<int> numbers) {
    for (int i = 0; i < numbers.size() - 1; i++) {
        int index_smallest_number = i;
        for (int j = i + 1; j < numbers.size(); j++) {
            if (numbers[j] < numbers[index_smallest_number]) {
                index_smallest_number = j;
            }
        }
        if (index_smallest_number != i) {
            std::swap(numbers[i], numbers[index_smallest_number]);
        }
    }
}

int main() {
    std::vector<int> numbers = {2, -2, 10, 0};
    selection_sort(numbers);
    for (auto number : numbers) {
        std::cout << number << " ";
    }
    std::cout << std::endl;
    return 0;
}